src/lasers/laser.c

```diff
@@ -82,14 +75,20 @@ Laser *create_laserline(cmplx pos, cmplx dir, float charge, float dur, const Col
 }
 
 Laser *create_laserline_ab(cmplx a, cmplx b, float width, float charge, float dur, const Color *clr) {
-	cmplx m = (b-a)*0.005;
-
-	return create_laser(a, 200, dur, clr, las_linear, static_laser, m, charge + I*width, 0, 0);
+	float lt = 200;
+	cmplx m = (b - a) / lt;
+	Laser *l = create_laser(a, 200, dur, clr, las_linear, m, 0, 0, 0);
+	INVOKE_TASK(laser_charge,
+		.laser = ENT_BOX(l),
+		.charge_delay = charge,
+		.target_width = width,
+	);
+	return l;
 }
 
 void laserline_set_ab(Laser *l, cmplx a, cmplx b) {
 	l->pos = a;
-	l->args[0] = (b - a) * 0.005;
```

Comment:
wait what was going on here?

Reply:
See `create_laserline_ab`:

```c
Laser *create_laserline_ab(cmplx a, cmplx b, float width, float charge, float dur, const Color *clr) {
	float lt = 200;
	cmplx m = (b - a) / lt;
	Laser *l = create_laser(a, 200, dur, clr, las_linear, m, 0, 0, 0);
	INVOKE_TASK(laser_charge,
		.laser = ENT_BOX(l),
		.charge_delay = charge,
		.target_width = width,
	);
	return l;
}
```

200 was chosen somewhat arbitrarily (by you) for the lifetime parameter here a long time ago, and 0.005 is the reciprocal of it. I made the `laserline_set_ab` function a little more general by taking the actual lifetime into account rather than hardcoding it.

Also, now that you brought it to my attention, I noticed that I forgot to replace one instance of 200 with lt in `create_laserline_ab`.